contracts/drafts/ERC20Migrator.sol

```diff
@@ -0,0 +1,77 @@
+pragma solidity ^0.4.24;
+
+import "../token/ERC20/IERC20.sol";
+import "../token/ERC20/ERC20Mintable.sol";
+import "../token/ERC20/SafeERC20.sol";
+import "../math/Math.sol";
+
+
+/**
+ * @title ERC20Migrator
+ * @dev This contract can be used to to migrate an ERC20 token from one
+ * contract to another, where each token holder has to opt-in to the migration.
+ * To opt-in, users must approve for this contract the number of tokens they
+ * want to migrate. Once the allowance is set up, anyone can trigger the
+ * migration to the new token contract. In this way, token holders "turn in"
+ * their old balance and will be minted an equal amount in the new token.
+ * The new token contract must be mintable. For the precise interface refer to
+ * OpenZeppelin's ERC20Mintable, but the only functions that are needed are
+ * `isMinter(address)` and `mint(address, amount)`. The migrator will check
+ * that it is a minter for the token.
+ * The balance from the legacy token will be transfered to the migrator, as it
+ * is migrated, and remain here forever.
+ *
+ * Although this contract can be used in many different scenarios, the main
+ * motivation was to provide a way of how an ERC20 token can be migrated to an
```

Comment:
Consider replacing _a way of how an ERC20 token can be migrated to an_ for _a way to migrate ERC20 tokens into an_